R/stan_module.R

```diff
@@ -117,6 +124,18 @@ setMethod(
     }
 )
 
+#' model_prep. Populates the model section of a StanModule
+#' @param x A StanModule object
+#' @export
+model_prep <- function(x) {
+    if (any(is.na(x@model), nchar(x@model) < 2)) {
```

Comment:
What about the case where model is populated ?  If i'm reading this right then it doesn't update the priors at all ? Do we not want to merge the priors in with the model ?

Reply:
If the user provides a stan code for the model section, I do not want to overwrite it.

Reply:
But does this mean if model code is provided the priors is just ignored ? 

I guess I was thinking it would be something like (pseudo code):

```
if not missing(priors):
   model = combine(priors, model)
```

Reply:
Well, I am not sure. I guess one of the two (priors or model) has to be ignored in the presence of the other. To my head, the user either specifies the prior list or he provides the complete stan code for the model section. I cannot see a way to "modify" a stan code given the prior list. I would assume that priority should be given to the stan code if provided.

Reply:
Hmmm I guess in my head I saw it working like this, like for example take the stan code:

```
model {
  sigma ~ lognormal(0, 0.5);         
  alpha ~ normal(0, sigma);       
  for (n in 1:N) {                        
    y[n] ~ bernoulli_logit(sum(alpha[team1[n]]) - sum(alpha[team0[n]]));
  }
}
```

Using our framework I imagine a valid way of splitting this code would be:

```
StanModule(
    priors = list(
        "sigma" = "lognormal(0, 0.5)",
        "alpha" = "normal(0, sigma)"
    ),
    model = c(
        "for (n in 1:N) {",
        "  y[n] ~ bernoulli_logit(sum(alpha[team1[n]]) - sum(alpha[team0[n]]));",
        "}"
    )
)
```

Likewise for the stanmodules which need that sum(log_lik) I imagine them being specified like:

```
StanModule(
    priors = list(
        "sigma" = "lognormal(0, 0.5)",
        "alpha" = "normal(0, sigma)"
    ),
    model = c(
        "target += sum(log_like)"
    )
)
```

As in the priors is an extension to the model which just provides some simple syntactic sugar

Reply:
It seems more structured than my suggestion. I will implement it to see how it works out.